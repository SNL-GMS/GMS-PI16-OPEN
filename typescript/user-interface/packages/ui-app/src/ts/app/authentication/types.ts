import { UserSessionTypes } from '@gms/ui-state';

export interface Authenticator {
  authenticateWith(userName: string): Promise<UserSessionTypes.AuthStatus>;
  checkIsAuthenticated(): Promise<UserSessionTypes.AuthStatus>;
  unAuthenticateWith(): Promise<UserSessionTypes.AuthStatus>;
  logout(setAuthStatus: (status: UserSessionTypes.AuthStatus) => void): void;
}

export interface LoginRequestParams {
  userName: string;
}
