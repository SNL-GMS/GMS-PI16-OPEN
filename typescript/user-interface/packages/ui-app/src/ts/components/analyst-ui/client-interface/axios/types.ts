/* eslint-disable @typescript-eslint/no-explicit-any */
import {
  ChannelSegmentTypes,
  EventTypes,
  QcMaskTypes,
  SignalDetectionTypes,
  WaveformTypes,
  WorkflowTypes
} from '@gms/common-model';
import { AxiosRequestConfig } from 'axios';
import { QueryResult } from 'react-query';

export interface EventMutationArgs {
  variables: {
    events: EventTypes.Event[];
  };
}

export interface QcMaskMutationArgs {
  variables: {
    qcMasks: QcMaskTypes.QcMask[];
  };
}

export interface SignalDetectionMutationArgs {
  variables: {
    signalDetections: SignalDetectionTypes.SignalDetection[];
  };
}

// ***************************************
// Workflow Interval Query and Mutation interfaces
// ***************************************
export interface StageIntervalsByStageIdAndTimeRequest {
  startTime: number;
  endTime: number;
  stageIds: WorkflowTypes.WorkflowDefinitionId[];
}
export interface UserRequest {
  userName: string;
  time: number;
}

export interface UpdateActivityIntervalStatusRequest extends UserRequest {
  activityIntervalId: WorkflowTypes.IntervalId;
  stageIntervalId: WorkflowTypes.IntervalId;
  status: WorkflowTypes.IntervalStatus;
}

export interface UpdateInteractiveAnalysisStageIntervalStatusRequest extends UserRequest {
  stageIntervalId: WorkflowTypes.IntervalId;
  status: WorkflowTypes.IntervalStatus;
}

export interface SplitActivityIntervalRequest {
  intervalId: WorkflowTypes.IntervalId;
  duration: number;
}
export interface EventQueryProps {
  eventQuery: QueryResult<EventTypes.Event[], any>;
}

export interface QcMaskQueryProps {
  qcMaskQuery: QueryResult<QcMaskTypes.QcMask[], any>;
}

export interface SignalDetectionQueryProps {
  signalDetectionQuery: QueryResult<SignalDetectionTypes.SignalDetection[], any>;
}

export interface WaveformQueryProps {
  waveformQuery: QueryResult<ChannelSegmentTypes.ChannelSegment<WaveformTypes.Waveform>[], any>;
}

/**
 * An axios request known to have a waveform query request in its data.
 */
export type WaveformAxiosRequestConfig = AxiosRequestConfig & {
  data: WaveformTypes.WaveformQueryArgs;
};
