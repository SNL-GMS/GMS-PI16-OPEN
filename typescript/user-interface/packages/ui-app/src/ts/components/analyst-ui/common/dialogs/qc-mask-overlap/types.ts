import { QcMaskTypes } from '@gms/common-model';

// !FIX ESLINT CIRCULAR DEPENDENCY
// eslint-disable-next-line import/no-cycle
import { CoordinatesPx } from '../qc-mask-form/types';
// !FIX ESLINT CIRCULAR DEPENDENCY
// eslint-disable-next-line import/no-cycle
import { QcMaskDialogBoxType } from '../types';

/**
 * QcMaskOverlap Props
 */
export interface QcMaskOverlapProps {
  masks: QcMaskTypes.QcMask[];
  // If set, any context menus opened as children will be created at given x/y coordinates
  contextMenuCoordinates?: CoordinatesPx;
  openNewContextMenu(
    x: number,
    y: number,
    mask: QcMaskTypes.QcMask,
    dialogType: QcMaskDialogBoxType
  );
  selectMask(mask: QcMaskTypes.QcMask);
}

/**
 * QcMaskOverlap  State
 */
export interface QcMaskOverlapState {
  selectedMask: QcMaskTypes.QcMask;
  createDialogOfType?: QcMaskDialogBoxType;
}
export interface QcMaskTableButtonParams {
  onClick(x: number, y: number, params: any);
}
