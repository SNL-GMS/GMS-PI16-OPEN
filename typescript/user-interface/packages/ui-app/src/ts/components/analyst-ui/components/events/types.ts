import { GQLTypes } from '@gms/common-graphql';
import { CommonTypes, EventTypes, WorkflowTypes } from '@gms/common-model';
import GoldenLayout from '@gms/golden-layout';
import { ChildProps, MutationFunction } from 'react-apollo';

/**
 * Types of events which can be show
 */
export enum EventFilters {
  EDGE = 'Edge',
  COMPLETED = 'Completed'
}

/**
 * Table row object for events
 */
export interface EventsRow {
  id: string;
  eventHypId: string;
  isOpen: boolean;
  stageId: string;
  lat: number;
  lon: number;
  depth: number;
  time: number;
  activeAnalysts: string[];
  numDetections: number;
  status: string;
  edgeEvent: boolean;
  signalDetectionConflicts: SignalDetectionConflict[];
}

export interface SignalDetectionConflict {
  stationName: string;
  id: string;
  arrivalTime: number;
  phase: CommonTypes.PhaseType;
}
/**
 * Event list local state
 */
export interface EventsState {
  currentTimeInterval: CommonTypes.TimeRange;
  suppressScrollOnNewData: boolean;
  showEventOfType: Map<EventFilters, boolean>;
}

/**
 * Mutations used in the event list
 */
export interface EventsMutations {
  updateEvents: MutationFunction;
  saveEvent: MutationFunction;
  saveAllModifiedEvents: MutationFunction;
}

/**
 * Props mapped in from Redux state
 */
export interface EventsReduxProps {
  // Passed in from golden-layout
  glContainer?: GoldenLayout.Container;
  currentTimeInterval: CommonTypes.TimeRange;
  analysisMode: WorkflowTypes.AnalysisMode;
  openEventId: string;
  selectedEventIds: string[];

  // callbacks
  setOpenEventId(
    event: EventTypes.Event | undefined,
    latestLocationSolutionSet: EventTypes.LocationSolutionSet | undefined,
    preferredLocationSolutionId: string | undefined
  ): void;
  setSelectedEventIds(ids: string[]): void;
}

/**
 * Consolidated props type for event list
 */
export type EventsProps = EventsReduxProps &
  ChildProps<EventsMutations> &
  GQLTypes.EventsInTimeRangeQueryProps &
  GQLTypes.DefaultStationsQueryProps &
  GQLTypes.SignalDetectionsByStationQueryProps &
  GQLTypes.WorkspaceStateProps;
