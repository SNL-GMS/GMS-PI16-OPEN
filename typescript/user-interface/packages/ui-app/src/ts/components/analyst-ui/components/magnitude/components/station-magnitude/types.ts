import { EventTypes } from '@gms/common-model';
import { AnalystWorkspaceTypes } from '@gms/ui-state';
import Immutable from 'immutable';
import { MutationFunction } from 'react-apollo';

// !FIX ESLINT CIRCULAR DEPENDENCY
// eslint-disable-next-line import/no-cycle
import { AmplitudesByStation } from '../../types';

/**
 * StationMagnitude State
 */
// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface StationMagnitudeState {
  computeNetworkMagnitudeSolutionStatus: Immutable.Map<
    string,
    [{ stationName: string; rational: string }]
  >;
}

export interface MagnitudeAndSdData {
  magSolution: EventTypes.NetworkMagnitudeSolution;
  sdData: StationMagnitudeSdData;
}

/**
 * options that can be passed to ag grid
 */
export interface Options {
  alignedGrids: any[];
}

/**
 * StationMagnitude Props
 */
export interface StationMagnitudeProps {
  options?: Options;
  amplitudesByStation: AmplitudesByStation[];
  historicalMode: boolean;
  selectedSdIds: string[];
  locationSolution: EventTypes.LocationSolution;
  displayedMagnitudeTypes: AnalystWorkspaceTypes.DisplayedMagnitudeTypes;
  computeNetworkMagnitudeSolution: MutationFunction;
  openEventId: string;

  setSelectedSdIds(ids: string[]): void;
  checkBoxCallback(
    magnitudeType: EventTypes.MagnitudeType,
    stationNames: string[],
    defining: boolean
  ): Promise<[{ stationName: string; rational: string }]>;
}

export interface StationMagnitudeSdData {
  channel: string;
  phase: string;
  amplitudeValue: number;
  amplitudePeriod: number;
  signalDetectionId: string;
  time: number;
  stationName: string;
  flagForReview: boolean;
}

export interface MagnitudeDataForRow {
  channel: string;
  signalDetectionId: string;
  phase: string;
  amplitudeValue: number;
  amplitudePeriod: number;
  flagForReview: boolean;
  defining: boolean;
  mag: number;
  res: number;
  hasMagnitudeCalculationError: boolean;
  computeNetworkMagnitudeSolutionStatus: string;
}

/**
 * Table row object for station magnitude
 */
export interface StationMagnitudeRow {
  id: string;
  dataForMagnitude: Map<EventTypes.MagnitudeType, MagnitudeDataForRow>;
  station: string;
  dist: number;
  azimuth: number;
  selectedSdIds: string[];
  historicalMode: boolean;
  azimuthTooltip: string;
  checkBoxCallback(
    magnitudeType: EventTypes.MagnitudeType,
    stationNames: string[],
    defining: boolean
  ): Promise<void>;
}
export interface StationMagAndSignalDetection {
  stationName: string;
  magnitudeAndSdData: Map<EventTypes.MagnitudeType, MagnitudeAndSdData>;
}
