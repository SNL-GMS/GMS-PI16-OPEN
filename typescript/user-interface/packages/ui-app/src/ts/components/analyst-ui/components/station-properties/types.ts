import { ChannelTypes, StationTypes } from '@gms/common-model';
import GoldenLayout from '@gms/golden-layout';
import {
  CellRendererParams,
  ColumnDefinition,
  Row,
  RowClickedEvent,
  ToolbarTypes
} from '@gms/ui-core-components';

export interface StationPropertiesComponentProps {
  // passed in from golden-layout
  glContainer?: GoldenLayout.Container;
}
export interface StationPropertiesPanelProps {
  selectedStation: string;
  effectiveAtTimes: string[];
}
export interface StationPropertiesToolbarProps {
  selectedEffectiveAt: string;
  selectedStation: StationTypes.Station;
  stationName: string;
  onEffectiveTimeChange: (args: any) => void;
  effectiveAtTimes: string[];
  widthPx?: number;
  setSelectedSiteColumnsToDisplay: (args: Map<siteColumnEnum, boolean>) => void;
  setSelectedChannelColumnsToDisplay: (args: Map<channelColumnEnum, boolean>) => void;
}

export interface SiteConfigurationTableProps {
  station: StationTypes.Station;
  onRowSelection(event: SiteConfigurationRowClickedEvent);
  columnsToDisplay: Map<string, boolean>;
}

export interface ChannelConfigurationTableProps {
  channels: ChannelTypes.Channel[];
  stationData: StationTypes.Station;
  columnsToDisplay: Map<string, boolean>;
}

export interface ChannelConfigurationRow extends Row {
  name: string;
  channelBandType;
  channelInstrumentType;
  channelOrientationType;
  channelOrientationCode;
  channelDataType;
  nominalSampleRateHz;
  description: string;
  calibrationFactor: string;
  calibrationStandardDeviation: string;
  calibrationPeriod: string;
  calibrationTimeShift: string;
  orientationHorizontalDegrees: string;
  orientationVerticalDegrees: string;
  latitudeDegrees: string;
  longitudeDegrees: string;
  depthKm: string;
  elevationKm: string;
  units: string;
  effectiveAt: string;
  effectiveUntil: string;
  calibrationEffectiveAt: string;
  fapResponseId: string;
  calibrationResponseId: string;
  northDisplacementKm: string;
  eastDisplacementKm: string;
  verticalDisplacementKm: string;
}

/**
 * Table row clicked event
 */
export type SiteConfigurationRowClickedEvent = RowClickedEvent<
  { id: string },
  any,
  number | string
>;

export interface SiteConfigurationRow extends Row {
  name: string;
  description: string;
  effectiveAt: string;
  effectiveUntil: string;
  latitudeDegrees: string;
  longitudeDegrees: string;
  elevationKm: string;
  depthKm: string;
  type: string;
}
export interface PropertiesToolbarItemDefs {
  rightToolbarItemDefs: ToolbarTypes.ToolbarItem[];
  leftToolbarItemDefs: ToolbarTypes.ToolbarItem[];
}

export type StationPropertiesColumnDefinition = ColumnDefinition<
  any, // RowDataType
  any, // ContextDataType
  string, // CellValueType
  any, // CellRendererParamsType
  any // HeaderRendererParamsType
>;

export type StationPropertiesCellRendererProps = CellRendererParams<
  SiteConfigurationRow | ChannelConfigurationRow, // RowDataType
  any, // ContextDataType
  string, // CellValueType
  any, // CellRendererParamsType
  {
    // HeaderRendererParamsType
    value: string | number;
    formattedValue: string;
  }
>;

/**
 * used to populate the values of the site column picker dropdown, and match the values to the table column ids
 */
export enum siteColumnEnum {
  name = 'name',
  effectiveAt = 'effectiveAt',
  effectiveUntil = 'effectiveUntil',
  latitudeDegrees = 'latitudeDegrees',
  longitudeDegrees = 'longitudeDegrees',
  depthKm = 'depthKm',
  elevationKm = 'elevationKm',
  description = 'description',
  type = 'type'
}

/**
 * used to match the display strings to values in the site table column picker dropdown
 */
export const siteColumnDisplayStrings: Map<string, string> = new Map<string, string>([
  ['name', 'Name'],
  ['effectiveAt', 'Effective At'],
  ['effectiveUntil', 'Effective Until'],
  ['latitudeDegrees', 'Latitude'],
  ['longitudeDegrees', 'Longitude'],
  ['depthKm', 'Depth'],
  ['elevationKm', 'Elevation'],
  ['description', 'Description'],
  ['type', 'Type']
]);

/**
 * used to match the display strings to values in the channel table column picker dropdown
 */
export const channelColumnDisplayStrings: Map<string, string> = new Map<string, string>([
  ['name', 'Name'],
  ['effectiveAt', 'Effective At'],
  ['effectiveUntil', 'Effective Until'],
  ['latitudeDegrees', 'Latitude'],
  ['longitudeDegrees', 'Longitude'],
  ['depthKm', 'Depth'],
  ['elevationKm', 'Elevation'],
  ['nominalSampleRateHz', 'Sample Rate'],
  ['units', 'Units'],
  ['orientationHorizontalDegrees', 'Horizontal Angle'],
  ['orientationVerticalDegrees', 'Vertical Angle'],
  ['calibrationFactor', 'Calibration Factor'],
  ['calibrationPeriod', 'Calibration Period'],
  ['calibrationEffectiveAt', 'Calibration Effective At'],
  ['calibrationTimeShift', 'Calibration Time Shift'],
  ['calibrationStandardDeviation', 'Calibration Std Dev'],
  ['northDisplacementKm', 'North Displacement'],
  ['eastDisplacementKm', 'East Displacement'],
  ['verticalDisplacementKm', 'Vertical Displacement'],
  ['description', 'Description'],
  ['channelDataType', 'Data Type'],
  ['channelBandType', 'Band Type'],
  ['channelInstrumentType', 'Instrument Type'],
  ['channelOrientationCode', 'Orientation Code'],
  ['channelOrientationType', 'Orientation Type'],
  ['calibrationResponseId', 'Calibration ID'],
  ['fapResponseId', 'FAP Resp ID']
]);

/**
 * used to populate the values of the channel column picker dropdown, and match the values to the table column ids
 */
export enum channelColumnEnum {
  name = 'name',
  effectiveAt = 'effectiveAt',
  effectiveUntil = 'effectiveUntil',
  latitudeDegrees = 'latitudeDegrees',
  longitudeDegrees = 'longitudeDegrees',
  depthKm = 'depthKm',
  elevationKm = 'elevationKm',
  nominalSampleRateHz = 'nominalSampleRateHz',
  units = 'units',
  orientationHorizontalDegrees = 'orientationHorizontalDegrees',
  orientationVerticalDegrees = 'orientationVerticalDegrees',
  calibrationFactor = 'calibrationFactor',
  calibrationPeriod = 'calibrationPeriod',
  calibrationEffectiveAt = 'calibrationEffectiveAt',
  calibrationTimeShift = 'calibrationTimeShift',
  calibrationStandardDeviation = 'calibrationStandardDeviation',
  northDisplacementKm = 'northDisplacementKm',
  eastDisplacementKm = 'eastDisplacementKm',
  verticalDisplacementKm = 'verticalDisplacementKm',
  description = 'description',
  channelDataType = 'channelDataType',
  channelBandType = 'channelBandType',
  channelInstrumentType = 'channelInstrumentType',
  channelOrientationCode = 'channelOrientationCode',
  channelOrientationType = 'channelOrientationType',
  calibrationResponseId = 'calibrationResponseId',
  fapResponseId = 'fapResponseId'
}
