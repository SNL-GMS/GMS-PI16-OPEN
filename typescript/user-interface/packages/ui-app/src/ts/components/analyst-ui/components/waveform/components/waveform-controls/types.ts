import { CommonTypes } from '@gms/common-model';
import { AnalystWorkspaceTypes } from '@gms/ui-state';
import React from 'react';

import { QcMaskDisplayFilters } from '~analyst-ui/config';
import { MaskDisplayFilter } from '~analyst-ui/config/user-preferences';

import { AlignWaveformsOn, PanType } from '../../types';
import { AmplitudeScalingOptions, FixedScaleValue } from './scaling-options';

/**
 * Waveform Display Controls Props
 */
export interface WaveformControlsProps {
  defaultSignalDetectionPhase: CommonTypes.PhaseType;
  currentSortType: AnalystWorkspaceTypes.WaveformSortType;
  currentOpenEventId: string;
  analystNumberOfWaveforms: number;
  showPredictedPhases: boolean;
  maskDisplayFilters: QcMaskDisplayFilters;
  alignWaveformsOn: AlignWaveformsOn;
  phaseToAlignOn: CommonTypes.PhaseType | undefined;
  alignablePhases: CommonTypes.PhaseType[] | undefined;
  isMeasureWindowVisible: boolean;
  measurementMode: AnalystWorkspaceTypes.MeasurementMode;
  setMode(mode: AnalystWorkspaceTypes.WaveformDisplayMode): void;
  setDefaultSignalDetectionPhase(phase: CommonTypes.PhaseType): void;
  setSelectedSortType(sortType: AnalystWorkspaceTypes.WaveformSortType): void;
  setAnalystNumberOfWaveforms(value: number, valueAsString?: string): void;
  setMaskDisplayFilters(key: string, maskDisplayFilter: MaskDisplayFilter): void;
  setWaveformAlignment(
    alignWaveformsOn: AlignWaveformsOn,
    phaseToAlignOn: CommonTypes.PhaseType,
    showPredictedPhases: boolean
  ): void;
  toggleMeasureWindow(): void;
  setShowPredictedPhases(showPredicted: boolean): void;
  pan(panDirection: PanType): void;
  onKeyPress(
    e: React.KeyboardEvent<HTMLDivElement>,
    clientX?: number,
    clientY?: number,
    channelId?: string,
    timeSecs?: number
  ): void;
  amplitudeScaleOption: AmplitudeScalingOptions;
  fixedScaleVal: FixedScaleValue;
  setAmplitudeScaleOption: (option: AmplitudeScalingOptions) => void;
  setFixedScaleVal: (val: FixedScaleValue) => void;
}

export interface WaveformDisplayControlsState {
  hasMounted: boolean;
}
