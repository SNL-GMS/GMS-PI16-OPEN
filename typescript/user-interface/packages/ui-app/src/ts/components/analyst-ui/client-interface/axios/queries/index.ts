import * as EventQuery from './event-query';
import * as QcMaskQuery from './qc-mask-query';
import * as SignalDetectionQuery from './signal-detection-query';
import * as WaveformQuery from './waveform-query';
import * as WorkflowIntervalQuery from './workflow-interval-query';
import * as WorkflowQuery from './workflow-query';
import * as WorkflowQueryUtil from './workflow-query-util';

export {
  EventQuery,
  QcMaskQuery,
  SignalDetectionQuery,
  WaveformQuery,
  WorkflowIntervalQuery,
  WorkflowQuery,
  WorkflowQueryUtil
};
