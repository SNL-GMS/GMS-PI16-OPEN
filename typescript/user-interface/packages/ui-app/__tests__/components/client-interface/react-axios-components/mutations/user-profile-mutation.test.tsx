import { UserProfileTypes } from '@gms/common-model';
import Axios from 'axios';
import * as Enzyme from 'enzyme';
import * as React from 'react';

import {
  setUserProfile,
  userProfileMutationConfig,
  useUserProfileMutation,
  withUserProfileMutation
} from '../../../../../src/ts/components/client-interface/axios/mutations/user-profile-mutation';
import { queryCache } from '../../../../../src/ts/components/client-interface/axios/queries/query-util';
import { UserProfileMutationArgs } from '../../../../../src/ts/components/client-interface/axios/types';
import {
  expectThatCompositionInjectsAProp,
  expectThatMutationHookMakesAxiosCall
} from '../../../../utils/query-mutation-test-utils';

// eslint-disable-next-line no-console
console.error = jest.fn();
Date.now = jest.fn().mockReturnValue(() => 1000);

const success = 'success';
Axios.request = jest.fn().mockReturnValue(Promise.resolve(success));

describe('User profile mutation mutation', () => {
  it('userProfileMutationConfig is defined', () => {
    expect(userProfileMutationConfig).toBeDefined();
  });

  const testUserProfile: UserProfileTypes.UserProfile = {
    audibleNotifications: [],
    defaultAnalystLayoutName: 'foo',
    defaultSohLayoutName: 'bar',
    userId: 'fooman',
    workspaceLayouts: []
  };

  const mutationArgs: UserProfileMutationArgs = {
    variables: {
      userProfile: testUserProfile
    }
  };

  it('userProfileMutationConfig on success invalidates the user profile query to force a re-fetch', async () => {
    queryCache.invalidateQueries = jest.fn();

    const prevNumCalls = (queryCache.invalidateQueries as jest.Mock).mock.calls.length;
    await userProfileMutationConfig.onSuccess();
    const { calls } = (queryCache.invalidateQueries as jest.Mock).mock;
    // check for the second argument of the call, to make sure it was called with the newly provided data.
    expect(calls).toHaveLength(prevNumCalls + 1);
  });

  it('setUserProfile makes a call to the server with the expected data', () => {
    // eslint-disable-next-line @typescript-eslint/no-floating-promises, jest/valid-expect-in-promise, jest/valid-expect
    expect(setUserProfile(mutationArgs))
      .resolves.toBe(success)
      // eslint-disable-next-line no-console
      .catch(e => console.error(e));
  });

  it('creates a wrapped component that matches a snapshot', () => {
    // eslint-disable-next-line @typescript-eslint/ban-types
    const TestComponent: React.FC<{}> = () => <div>test</div>;
    const wrapperFn = withUserProfileMutation();
    const WrappedComponent = wrapperFn(TestComponent);
    const wrapper = Enzyme.mount(<WrappedComponent />);
    expect(wrapper).toMatchSnapshot();
  });

  // eslint-disable-next-line jest/expect-expect
  it('can inject a setUserProfile prop into a wrapped component', () => {
    expectThatCompositionInjectsAProp(withUserProfileMutation, 'setUserProfile');
  });

  // eslint-disable-next-line jest/expect-expect
  it('makes a request to the server when the mutation is called', () => {
    expectThatMutationHookMakesAxiosCall(useUserProfileMutation, mutationArgs);
  });
});
