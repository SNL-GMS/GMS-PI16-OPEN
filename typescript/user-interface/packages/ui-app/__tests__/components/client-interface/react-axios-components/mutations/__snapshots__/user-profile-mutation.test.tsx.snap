// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User profile mutation mutation creates a wrapped component that matches a snapshot 1`] = `
<Component>
  <ComponentWithMutation>
    <TestComponent
      setUserProfile={[Function]}
    >
      <div>
        test
      </div>
    </TestComponent>
  </ComponentWithMutation>
</Component>
`;
